### PR TITLE
Make sure that cronfiles are also filtered with expandproperties if run under sudo

### PR DESCRIPTION
**Pull request by RobQuistNL**
We noticed that cronfiles copied using the sudo method were not replace (which is rather logical).

This way they are first copied to a temporary directory in order to replace the variables in the files, then copied from there to the cron directory under sudo user.

@ngroot :)

### CODE REVIEW

File: lib/cron/cron.xml [hunk 1]
```diff
@@ -39,7 +42,18 @@
     </target>
 
     <target name="cron-copy-files-sudo">
-        <exec command="sudo cp ${cron.files}${project.app}-${project.environment}-* ${cron.location}" logoutput="true"/>
+        <copy todir="/tmp/phing-cron-copy-${TIMESTAMP}">
```
Comment (jvandijk) [inline lib/cron/cron.xml:45]: Why so difficult? Can't the files be expanded in place... and then copied over?

Comment (RobQuistNL) [inline lib/cron/cron.xml:45]: Why would you want that? That doesn't feel right... It means changing original files in the repo :/
I know we do it for certain config files... But this is somewhat different if you ask me. Maybe the cron files will be read out by the application or something? I don't know.

Comment (jvandijk) [inline lib/cron/cron.xml:45]: Phing is running when either the files or copied to the server, or in the case of our project rsynced to a different 'release' directory. So... No, we are are not editing files in the repo if you ask me.

Comment (RobQuistNL) [inline lib/cron/cron.xml:45]: With the word "repo" i meant the deployed version. We're editing data there. If you'd need the pure cron file within your project, you would open up `${cron.files}`. If you want the replaced / filled cron files, you'd have to parse them yourself or open up `${cron.location}`. I'm just not a big fan of editing those files inside the deployed version. I wonder what your opinion on this is, @ngroot :)

Comment (jvandijk) [inline lib/cron/cron.xml:45]: The deployed version should be unaware of any deployment tool at ALL! Hence the usage of tools like phing, capistrano and such to set environment variables. The same counts for these files. I really don't see any argument for your consideration that you're not a big fan of editing files in the deployed version.

@nickbelhomme @david4worx what are your opinions?

Comment (NickBelhomme) [inline lib/cron/cron.xml:45]: IMHO you are not working on the repo (shared-folder) but on the deployed files (release folder). And there is nothing wrong in letting capistrano prepare these files through phing. Capistrano is meant for setting stuff in the deploy location (release-dir) such as these vars. I rather have these cron files prepped by capistrano in the release dir, so they are configured correctly (pure) to be potentially used by your project than copy them over to some tmp dir where all context is lost.

edit in release dir  +1
copy to /tmp -1

**Comment by RobQuistNL** [comment]
OK, more votes for that solution, i'll do that.

**Comment by RobQuistNL** [comment]
I've been searching but can't really find anything for this.

I've tried;
replacetokens - Don't exist in phing
replace - Doesn't exist either

```
<target name="cron-replace-params">
        <replaceregexp match="\${(.*?)}" replace="${\1}">
            <fileset dir="${cron.files}">
                <include name="${project.app}-${project.environment}-*" />
            </fileset>
        </replaceregexp>
    </target>
```

Seems to not parse the token in the replace path - thus not actually doing anything.

Is there any other way to replace all tokens in files that you know of? Otherwise this actually has to be the solution.

Or, copy to tmp dir with filterchain, copy back to repo, then execute other phing tasks.. @NickBelhomme  @jvandijk

**Comment by NickBelhomme** [comment]
I would go for this solution: https://github.com/CrazyAwesomeCompany/autophing/pull/7
This PR can be closed.

**Pull request closed without merging**